drivers/video/gc2145.c

```diff
@@ -0,0 +1,1145 @@
+			    uint16_t reg_num)
+{
+	uint16_t i = 0;
+	const struct gc2145_config *cfg = dev->config;
+
+	for (i = 0; i < reg_num; i++) {
+		int err;
+
+		err = gc2145_write_reg(&cfg->i2c, regs[i].addr, regs[i].value);
+		if (err) {
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static int gc2145_soft_reset(const struct device *dev)
+{
+	int ret = 0;
+	const struct gc2145_config *cfg = dev->config;
+
+	/* Initiate system reset */
+	ret = gc2145_write_reg(&cfg->i2c, GC2145_REG_RESET, GC2145_REG_SW_RESET);
+
+	k_msleep(30);
+
+	return ret;
+}
+
+static int gc2145_set_ctrl_vflip(const struct device *dev, int enable)
+{
+	int ret;
+	const struct gc2145_config *cfg = dev->config;
+	uint8_t old_value = 0;
+
+	ret = gc2145_read_reg(&cfg->i2c, GC2145_REG_AMODE1, &old_value);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Set the vertical flip state */
+	return gc2145_write_reg(&cfg->i2c, GC2145_REG_AMODE1, (old_value & 0xFC) | (enable << 1));
+}
+
+static int gc2145_set_ctrl_hmirror(const struct device *dev, int enable)
+{
+	int ret;
+	const struct gc2145_config *cfg = dev->config;
+	uint8_t old_value = 0;
+
+	ret = gc2145_read_reg(&cfg->i2c, GC2145_REG_AMODE1, &old_value);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Set the horizontal mirror state */
+	return gc2145_write_reg(&cfg->i2c, GC2145_REG_AMODE1, (old_value & 0xFC) | enable);
+}
+
+static int gc2145_set_window(const struct device *dev, uint16_t reg, uint16_t x, uint16_t y,
+			     uint16_t w, uint16_t h)
+{
+	int ret;
+	const struct gc2145_config *cfg = dev->config;
+
+	ret = gc2145_write_reg(&cfg->i2c, 0xFE, GC2145_SET_P0_REGS);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Y/row offset */
+	ret = gc2145_write_reg(&cfg->i2c, reg++, y >> 8);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gc2145_write_reg(&cfg->i2c, reg++, y & 0xff);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* X/col offset */
+	ret = gc2145_write_reg(&cfg->i2c, reg++, x >> 8);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gc2145_write_reg(&cfg->i2c, reg++, x & 0xff);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Window height */
+	ret = gc2145_write_reg(&cfg->i2c, reg++, h >> 8);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gc2145_write_reg(&cfg->i2c, reg++, h & 0xff);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Window width */
+	ret = gc2145_write_reg(&cfg->i2c, reg++, w >> 8);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gc2145_write_reg(&cfg->i2c, reg++, w & 0xff);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return ret;
+}
+
+static int gc2145_set_output_format(const struct device *dev, int output_format)
+{
+	int ret;
+	const struct gc2145_config *cfg = dev->config;
+
+	ret = gc2145_write_reg(&cfg->i2c, 0xFE, GC2145_SET_P0_REGS);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (output_format == VIDEO_PIX_FMT_RGB565) {
+		/* Disable JPEG compression and set output to RGB565 */
+		ret = gc2145_write_reg(&cfg->i2c, GC2145_REG_OUTPUT_FMT,
+				       GC2145_REG_OUTPUT_FMT_RGB565);
+	} else {
+		LOG_ERR("Image format not supported");
+		return -ENOTSUP;
+	}
```

Comment:
swap the if, return in a branch do the normal execution inline, have the function end with `return 0`